BJS2-77048 Анализ лайков: Consumer

VladimirEkgardt opened the pull request.

VladimirEkgardt committed:
BJS2-77048 Анализ лайков: Consumer

VladimirEkgardt committed:
BJS2-77048 Анализ лайков: Consumer

pirojok-yt commented on src/main/java/faang/school/analytics/config/KafkaConsumerConfig.java:
```diff
@@ -0,0 +1,53 @@
+                new ConcurrentKafkaListenerContainerFactory<>();
+
+        factory.setConsumerFactory(consumerFactory());
+        factory.setConcurrency(3);
```
можно 3 в константу вынести

pirojok-yt commented on src/main/java/faang/school/analytics/listener/EventListener.java:
```diff
@@ -0,0 +1,63 @@
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class EventListener {
```
я думаю лучше назвать AnalyticsEventListener

VladimirEkgardt replied:
Ну я предполагал что-то типа считыватель эвентов, так как через этот класс будут проходить любые эвенты

pirojok-yt replied:
сорри, я не так понял сам класс

VladimirEkgardt committed:
Вынес переменную в пропертис

VladimirEkgardt committed:
1) KafkaConsumerConfig: изменил дефолтное значение десериализации на дто
2) PostLikeEventListener: изменил поле consume на дто
3) EventListener: убрал ручной маппер в дто
4) Исправил тесты

SnurnicynKonstantin commented on src/main/resources/application.yaml:
```diff
@@ -34,3 +35,13 @@ logging:
+kafka:
+  host: "localhost:9092"
+  topic-name: "like"
+  group: "likes"
```
У этого сервиса лучше использовать имя сервиса в названии группы. Например analytic-group

SnurnicynKonstantin commented on src/main/resources/application.yaml:
```diff
@@ -34,3 +35,13 @@ logging:
+
+kafka:
+  host: "localhost:9092"
+  topic-name: "like"
```
И зесь лучше создать раздел topics: и уже в нем like-post:.
А то аналитика будет работать с многими топиками. И когда будем добавлять друкие листенеры, то для них то же придется прописывать топики

SnurnicynKonstantin commented on src/main/java/faang/school/analytics/listener/PostLikeEventListener.java:
```diff
@@ -0,0 +1,26 @@
+    private String topicName;
+
+    @Value("${kafka.group}")
+    private String groupName;
```
Зачем эти переменные, если ты на 22 и так из пропертей берешь значения?

SnurnicynKonstantin reviewed (commented):
В остальном все отрабатывает четко. Молодец)

VladimirEkgardt committed:
1)Application.yaml: добавил раздел topics и groups
2) Переименовал путь переменных в KafkaConfig и PostLikeEventListener
